app/Http/Controllers/MonitorController.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use DateTime;
+use DateInterval;
+use DatePeriod;
+use App\Enums\ClassicPalette;
+use App\Enums\HighContrastPalette;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+use stdClass;
+use Symfony\Component\HttpFoundation\Response;
+use Illuminate\Http\Request;
+
+class MonitorController extends AbstractController
+{
+    /**
+     * Get statistics about our recent submissions queue performance.
+     */
+    public function get(): JsonResponse
+    {
+        $user = Auth::user();
+        if ($user === null || !$user->IsAdmin()) {
+            return response()->json(['error' => 'Permissions error'], status: Response::HTTP_FORBIDDEN);
+        }
+
+        // Get the length of the current backlog and how long the oldest submission
+        // has been waiting.
+        $backlog_length = DB::table('jobs')->count();
+        $backlog_time = null;
+        if ($backlog_length > 0) {
+            $backlog_time = time_difference(time() - DB::table('jobs')->min('created_at'), false, 'ago');
+        }
+
+        // Choose colors for the chart based on user preferences.
+        if (array_key_exists('colorblind', $_COOKIE) && intval($_COOKIE['colorblind']) === 1) {
+            $palette = HighContrastPalette::class;
+        } else {
+            $palette = ClassicPalette::class;
+        }
+
+        // Initialize trendline data.
+        $success_values = [];
+        $fail_values = [];
+        $ticks = [];
+        $period = new \DatePeriod(
+            new DateTime('24 hours ago'),
+            new DateInterval('PT1H'),
+            24);
+        $i = 0;
+        foreach ($period as $datetime) {
+            // Truncate this timestamp to the beginning of the hour.
+            $timestamp = $datetime->setTime(intval($datetime->format('H')), 0, 0)->getTimestamp();
+            $success_values[$timestamp] = 0;
+            $fail_values[$timestamp] = 0;
+            $i++;
+            if ($i % 3 === 0) {
+                $ticks[] = $timestamp;
+            }
+        }
+
+        // Populate pass/fail trendline data.
+        foreach ($this->resultsPerHour('failed_jobs', 'failed_at') as $row) {
+            $key = $row->truncated_time;
+            if (array_key_exists($key, $fail_values)) {
+                $fail_values[$key] += $row->n_jobs;
```

Comment:
Is it possible for this condition to happen now?

Reply:
I'm tempted to keep these checks "just in case" to avoid array key error / HTTP 500 from the API endpoint. One possible failure case if it the "daily updates" job fails to run, resulting in old submission records. I think I did a good job mollifying timezone / daylight saving time shenanigans, but that's always a linger fear of mine too.